### PR TITLE
Fixes #9356 - Add about_Data_Files topic

### LINKED ISSUE #9356: Official Documentation on the .psd1 Powershell Data File Format
### Prerequisites

- [X] **Existing Issue:** Search the existing issues for this repository. If there is an issue that fits your needs do not file a new one. Subscribe, react, or comment on that issue instead.
- [X] **Descriptive Title:** Write the title for this issue as a short synopsis. If possible, provide context. For example, "Document new `Get-Foo` cmdlet" instead of "New cmdlet."

### PowerShell Version

7.3

### Summary

Besides the description of how a module manifest has to look like, there seems to be hardly any documentation on how exactly the powershell data file needs to be formatted, and what mode is used when the files are loaded.

Trigger for this ticket was a thread in the powershell discord: https://discord.com/channels/180528040881815552/1034501153947385856

### Details

Please write some elaborate documentation for data files and how exactly the data is handled when they are imported using the several commands like Import-PowershellDataFile or Import-LocalizedData

### Proposed Content Type

About Topic

### Proposed Title

about_PowerShell_Data_Files

### Related Articles

- https://discord.com/channels/180528040881815552/1034501153947385856
- https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/import-powershelldatafile?view=powershell-7.2
- https://learn.microsoft.com/powershell/module/microsoft.powershell.core/about/about_language_modes#restricted-language-restrictedlanguage
- https://devblogs.microsoft.com/powershell/powershell-constrained-language-mode/
- http://ramblingcookiemonster.github.io/PowerShell-Configuration-Data/

### PR DESCRIPTION
**Pull request by sdwheeler**
# PR Summary

- Fixes [AB#59982](https://dev.azure.com/msft-skilling/cebd7ef5-4282-448b-9701-88c8637581b7/_workitems/edit/59982) - Fixes #9356 

## PR Checklist

- [x] **Descriptive Title:** This PR's title is a synopsis of the changes it proposes.
- [x] **Summary:** This PR's summary describes the scope and intent of the change.
- [x] **Contributor's Guide:** I have read the [contributors guide][contrib].
- [x] **Style:** This PR adheres to the [style guide][style].

[contrib]: https://learn.microsoft.com/powershell/scripting/community/contributing/overview
[style]: https://learn.microsoft.com/powershell/scripting/community/contributing/powershell-style-guide

### CODE REVIEW

**Comment by sdwheeler** [comment]
Retrigger the build

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Data_Files.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+The `Import-PowerShellDataFile` cmdlet imports basic data files. The data in
+the file must be contained in a hashtable. This format only supports constant
+values. You can't use code or PowerShell expressions.
+
+## Module manifests
+
+Module manifests are PowerShell data files. The data in the file must be
+contained in a hashtable. The structure of that hashtable only supports
+specific key name related to PowerShell modules.
```
Comment (michaeltlombardi) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Data_Files.md:30]:
```suggestion
specific key names related to PowerShell modules.
```

File: reference/7.2/Microsoft.PowerShell.Core/About/about_Data_Files.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+The `Import-PowerShellDataFile` cmdlet imports basic data files. The data in
+the file must be contained in a hashtable. This format only supports constant
+values. You can't use code or PowerShell expressions.
+
+## Module manifests
+
+Module manifests are PowerShell data files. The data in the file must be
+contained in a hashtable. The structure of that hashtable only supports
+specific key name related to PowerShell modules.
```
Comment (michaeltlombardi) [inline reference/7.2/Microsoft.PowerShell.Core/About/about_Data_Files.md:30]:
```suggestion
specific key names related to PowerShell modules.
```

File: reference/7.3/Microsoft.PowerShell.Core/About/about_Data_Files.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+The `Import-PowerShellDataFile` cmdlet imports basic data files. The data in
+the file must be contained in a hashtable. This format only supports constant
+values. You can't use code or PowerShell expressions.
+
+## Module manifests
+
+Module manifests are PowerShell data files. The data in the file must be
+contained in a hashtable. The structure of that hashtable only supports
+specific key name related to PowerShell modules.
```
Comment (michaeltlombardi) [inline reference/7.3/Microsoft.PowerShell.Core/About/about_Data_Files.md:30]:
```suggestion
specific key names related to PowerShell modules.
```

File: reference/7.4/Microsoft.PowerShell.Core/About/about_Data_Files.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+The `Import-PowerShellDataFile` cmdlet imports basic data files. The data in
+the file must be contained in a hashtable. This format only supports constant
+values. You can't use code or PowerShell expressions.
+
+## Module manifests
+
+Module manifests are PowerShell data files. The data in the file must be
+contained in a hashtable. The structure of that hashtable only supports
+specific key name related to PowerShell modules.
```
Comment (michaeltlombardi) [inline reference/7.4/Microsoft.PowerShell.Core/About/about_Data_Files.md:30]:
```suggestion
specific key names related to PowerShell modules.
```

**Review by michaeltlombardi — DISMISSED** [review]

**Comment by sdwheeler** [comment]
Retrigger the build

**Pull request merged**